core/oputil/oputil_ssh_key.js

```diff
@@ -0,0 +1,139 @@
+
+//	deps
+const async = require('async');
+const fs = require('fs-extra');
+const exec = require('child_process').exec;
+const inq = require('inquirer');
+const _ = require('lodash');
+
+
+exports.handleSSHKeyCommand = handleSSHKeyCommand;
+
+const ConfigIncludeKeys = [
+    'loginServers.ssh',
+    'loginServers.ssh.privateKeyPem',
+];
+
+const MINIMUM_PASSWORD_LENGTH = 8;
+const QUESTIONS = {
+    Create: [
+        {
+            name: 'createNew',
+            message: 'Generate New SSH Keys?',
+            type: 'confirm',
+            default: false,
+        },
+        {
+            name: 'password',
+            message: 'SSL Password:',
+            default: "",
+            when: answers => answers.createNew,
+        },
+    ],
+};
+
+function execute(ui, command) {
+    exec(
+        command,
+        function (error, stdout, stderr) {
+            ui.log.write(error);
+
+            if (error) {
+                const reason = error ? error.message : 'OpenSSL Error';
+                logDebug(
+                    {
+                        reason: reason,
+                        cmd: util.cmd,
+                        args: args
+                    },
+                    `openssl command failed`
+                );
+            }
+            else {
+                ui.log.write("SSH Keys Generated")
+            }
+        }
+    );
+}
+
+function createNew(cb) {
+    const ui = new inq.ui.BottomBar();
+
+    async.waterfall(
+        [
+            function init(callback) {
+                return initConfigAndDatabases(callback);
+            },
+            function create(configuration, callback) {
+                getAnswers(QUESTIONS.Create, answers => {
+                    if (!answers.createNew) {
+                        return callback('exit');
+                    }
+
+                    // Get Answer Value
+                    const sslPassword = answers.password.trim();
+                    if (!sslPassword || sslPassword == "") {
+                        ui.log.write('Password must be set.');
+
+                        return callback('exit');
+                    }
+                    if (sslPassword.length < MINIMUM_PASSWORD_LENGTH) {
+                        ui.log.write(`Password must be at least ${MINIMUM_PASSWORD_LENGTH} characters.`);
+
+                        return callback('exit');
+                    }
+
+                    // Check if Keyfiles Exist
+                    const sshKeyPath = "config/security/";
+                    const sshKeyFilename = "ssh_private_key.pem";
+                    const targetKeyFile = sshKeyPath + sshKeyFilename;
+
+                    ui.log.write(`Creating SSH Key: ${targetKeyFile}`);
+
+                    // Create Dir
+                    if (!fs.pathExists(sshKeyPath)) {
+                        ui.log.write(`Creating Directory: ${sshKeyPath}`);
+                        exec(`mkdir -p ${sshKeyPath}`);
```

Comment:
Sorry, I should have made more clear what I was pointing out: fs.ensureDirSync() for example vs execing() another process. That removes the need for checking if it exists, see previous link.

Reply:
Ah, that's much easier.

I mainly code in C#/C++, so I'm used to breaking everything into steps. We don't have a magical function that checks if exists, creates the dir, and gives you a hug.